test/testmachinery/shoot/enable_disable_test.go

```diff
@@ -0,0 +1,126 @@
+// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package shoot_test
+
+import (
+	"context"
+	"time"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	testutils "github.com/gardener/gardener/pkg/utils/test"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+	"github.com/gardener/gardener/test/framework"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gardener/gardener-extension-shoot-rsyslog-relp/test/common"
+)
+
+const (
+	defaultTestTimeout        = 30 * time.Minute
+	defaultTestCleanupTimeout = 10 * time.Minute
+)
+
+var _ = Describe("Shoot rsyslog-relp  testing", func() {
+	var (
+		f            = framework.NewShootFramework(nil)
+		echoServerIP string
+	)
+
+	test := func(parentCtx context.Context, shootMutateFn func(shoot *gardencorev1beta1.Shoot) error) {
+		By("Enable the shoot-rsyslog-relp extension")
+		ctx, cancel := context.WithTimeout(parentCtx, 10*time.Minute)
+		defer cancel()
+		Expect(f.UpdateShoot(ctx, shootMutateFn)).To(Succeed())
+
+		By("Verify shoot-rsyslog-relp works")
+		echoServerPodIf, echoServerPodName, err := common.GetEchoServerPodInterfaceAndName(ctx, f.ShootClient)
+		Expect(err).NotTo(HaveOccurred())
+		verifier := common.NewVerifier(f.Logger, f.ShootClient, echoServerPodIf, echoServerPodName, f.Project.Name, f.Shoot.Name, string(f.Shoot.UID))
+
+		common.ForEachNode(ctx, f.ShootClient, func(ctx context.Context, node *corev1.Node) {
+			verifier.VerifyExtensionForNode(ctx, node.Name)
+		})
+
+		By("Disable the shoot-rsyslog-relp extension")
+		Expect(f.UpdateShoot(ctx, func(shoot *gardencorev1beta1.Shoot) error {
+			common.RemoveRsyslogRelpExtension(shoot)
+			return nil
+		})).To(Succeed())
+
+		By("Verify that shoot-rsyslog-relp extension is disabled")
+		ctx, cancel = context.WithTimeout(parentCtx, 5*time.Minute)
+		DeferCleanup(cancel)
+		common.ForEachNode(ctx, f.ShootClient, func(ctx context.Context, node *corev1.Node) {
+			verifier.VerifyExtensionDisabledForNode(ctx, node.Name)
+		})
+	}
+
+	framework.CBeforeEach(func(ctx context.Context) {
+		By("Deploy rsyslog-relp-echo-server in Shoot cluster")
+		var err error
+		echoServerIP, err = createRsyslogRelpEchoServer(ctx, f)
+		Expect(err).NotTo(HaveOccurred())
+	}, time.Minute)
+
+	framework.CAfterEach(func(ctx context.Context) {
+		By("Delete rsyslog-relp-echo-server from Shoot cluster")
+		Expect(deleteRsyslogRelpEchoServer(ctx, f))
+	}, time.Minute)
+
+	FContext("shoot-rsyslog-relp extension with tls disabled", Label("tls-disabled"), func() {
```

Comment:
This should be a leftover from testing.
```suggestion
	Context("shoot-rsyslog-relp extension with tls disabled", Label("tls-disabled"), func() {
```